### ISSUE TITLE
llvm10 branch regression: dynamic_library standalone test

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#4388]
To reproduce:

 * Check out `llvm10` branch (you need llvm/clang/lld `release/10.x` branch built from source)

```
$ make
$ ./zig build test-standalone
Segmentation fault at address 0x7fcaa35113d0
/home/andy/tmp/zig/lib/std/dynamic_library.zig:239:32: 0x2368c3 in std.dynamic_library.ElfLib.lookup (main)
        while (i < self.hashtab[1]) : (i += 1) {
                               ^
The following command terminated unexpectedly:
cd /home/andy/tmp/zig/test/standalone/load_dynamic_library && /home/andy/tmp/zig/test/standalone/load_dynamic_library/zig-cache/o/RZXUl1BuHDFO5Rgu1o9Yhycy1g-meqa41XV1av2ZbCXCvh172895XhMfXK0icd2f/main /home/andy/tmp/zig/test/standalone/load_dynamic_library/zig-cache/o/v5hK0YaPwcVpBZPH48DbsQDD1QDav8DBywJ3Kg6dSU-v9QbTAOVSRLNmT0u9QGmt/libadd.so.1.0.0 

Build failed. The following command failed:
/home/andy/tmp/zig/test/standalone/load_dynamic_library/zig-cache/o/ROglZF87eFlbQBUonCFYFNbAPwgXZtF6w2D5Gb9Z1T80RGJ90W39yWMr95GaeA1q/build /home/andy/tmp/zig/build-llvm10-release/zig /home/andy/tmp/zig/test/standalone/load_dynamic_library /home/andy/tmp/zig/test/standalone/load_dynamic_library/zig-cache test
The following command exited with error code 1 (expected 0):
cd /home/andy/tmp/zig && build-llvm10-release/zig build --build-file /home/andy/tmp/zig/test/standalone/load_dynamic_library/build.zig test 

Build failed. The following command failed:
/home/andy/tmp/zig/zig-cache/o/F4bWsAjnUAOZsB9c7xU896UyJEU88zTBI_38Pu7av_E7t8cekbBAKdcfWWVPV2ru/build /home/andy/tmp/zig/build-llvm10-release/zig /home/andy/tmp/zig /home/andy/tmp/zig/zig-cache test-standalone
```

It's unclear what the problem is so far. Perhaps it is a bug in the std lib dynamic library loading code which could be fixed.

cc @LemonBoy I was kinda hoping you might have some insight here

**Comment by andrewrk** [comment]
> cc @LemonBoy I was kinda hoping you might have some insight here

Never mind that, it's just that the libc-less dynamic library loading is incomplete and needs more work.

**Comment by andrewrk** [comment]
Fixed in 5076f2d4f6918946616504d22907c9f80601caf8
